pennylane/devices/default_clifford.py

```diff
@@ -210,6 +251,64 @@ def circuit():
         Maintaining and working with this tableau representation instead of the complete state vector
         makes the calculations of increasingly large Clifford circuits more efficient on this device.
 
+    .. details::
+        :title: Probabilities for Basis States
+        :href: clifford-probabilities
+
+        As the ``default.clifford`` device supports executing quantum circuits with a large number of qubits,
+        we restrict the ability to compute the ``analytical`` probabilities for `all` computational basis
+        states at once to maintain computational efficiency when system size scales beyond ``20`` qubits.
```

Comment:
Is this still a hardcoded restriction? I don't think we need to check this, similar to how we don't check if a user tries to run a 100 qubit circuit in `default.qubit`.